### DISCUSSION TITLE
Where does the arrow plotting end?

### DISCUSSION THREAD

**Discussion by psads-git** [cetz-package/cetz#666] [Q&A] [answered]
The following code produces the figure below. Could you please explain why we apparently need to use different side paddings (`left: 2.8em`, `right: 1em`) to get the formula equally distant from both arrows? I suspect the arrow head is not the end coordinate when `cetz` plots the first arrow. Thanks!

```
#import "@preview/cetz:0.2.2"

#cetz.canvas({
  import cetz.draw: *

  line((0, 0), (2, 0), mark: (end: ">", fill: black), name: "line")
  content((rel: (1, 0)), align(center, $a^2 = b^2+c^2$), 
      padding: (left: 2.8em, right: 1em), name: "form")
  line("form.east", (rel: (2, 0)), mark: (end: ">", fill: black))
})
```

![image](https://github.com/user-attachments/assets/b6ca6225-2683-46ad-ad85-51ec5913a051)

**Comment by fenjalien** [comment, marked as answer]
The arrow is drawn at the end coordinate, its behaving like that because you're using relative coordinates to place the equation. You should use anchors to place it intead:
```
#import "@preview/cetz:0.2.2"

#cetz.canvas({
  import cetz.draw: *

  line((0, 0), (2, 0), mark: (end: ">", fill: black), name: "line")
  content(
    "line.end",
    align(center, $a^2 = b^2+c^2$), 
    padding: (left: 1em, right: 1em),
    name: "form",
    anchor: "west"
  )
  line("form.east", (rel: (2, 0)), mark: (end: ">", fill: black))
})
```
![image](https://github.com/user-attachments/assets/c984c86e-0709-48b0-8adb-2e07bc96a8c5)

**Comment by psads-git** [comment]
Thanks, @fenjalien, that works perfectly!
